server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java

```diff
@@ -154,6 +172,7 @@ public void startReplication(ActionListener<Void> listener) {
 
     private void getFiles(CheckpointInfoResponse checkpointInfo, StepListener<GetSegmentFilesResponse> getFilesListener)
         throws IOException {
+        cancellableThreads.checkForCancel();
```

Comment:
How is the case for partial files copy is handled i.e. replica copied few but not all files ? 
Can this be problematic when this replica is promoted as new primary with few files from previous primary ?

Reply:
If there are partial files copied they will remain tmp with a prefix like `replication_0.cfe`.  The rename occurs once all files have been copied in finalize.  So in this case the replication would be cancelled and the partial files discarded in the commit made before the shard's engine is re-opened swapped.